src/Microsoft.ML.AutoML/EstimatorExtensions/EstimatorExtensionCatalog.cs

```diff
@@ -37,7 +40,10 @@ internal class EstimatorExtensionCatalog
             { EstimatorName.OneHotEncoding, typeof(OneHotEncodingExtension) },
             { EstimatorName.OneHotHashEncoding, typeof(OneHotHashEncodingExtension) },
             { EstimatorName.TextFeaturizing, typeof(TextFeaturizingExtension) },
-            { EstimatorName.ImageFeaturizing, typeof(ImageFeaturizingExtension) },
+            { EstimatorName.ImageLoading, typeof(ImageLoadingExtension) },
+            { EstimatorName.ImageResizing, typeof(ImageResizingExtension) },
+            { EstimatorName.PixelExtracting, typeof(PixelExtractingExtension) },
+            { EstimatorName.ResNet18Featurizing, typeof(ResNet18FeaturizingExtension) },
```

Comment:
Can we have a DNNImageFeaturizer w/ a parameter of ResNet18? This will allow us to CodeGen once for the DNNImageFeaturizer and use it in multiple ways.

Reply:
I would prefer to do it separately. It might be a bit duplicate but will keep the corresponding CodeGen part simple and straightforward

Reply:
#resolved